File: 0238-product-of-array-except-self/solution.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] ans=new int[nums.length];
        Arrays.fill(ans,1);
        for(int i=nums.length-2;i>=0;i--){
            ans[i]=nums[i+1]*ans[i+1];
        }
        int p=nums[0];
        for(int i=1;i<nums.length-1;i++){
            ans[i]=p*ans[i];
            p*=nums[i];
        }
        ans[nums.length-1]=p;
        return ans;
    }
}

File: 0238-product-of-array-except-self/solution.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        ans=[]
        for i in range(0,len(nums)):
            ans.append(1)
        for i in range(0,len(nums)):
            if(i+1<len(nums)):
                ans[i+1]=nums[i]*ans[i]
        mul=nums[len(nums)-1]
        for i in range(len(ans)-2,-1,-1):
            ans[i]=mul*ans[i]
            mul*=nums[i]
        return ans

